### DISCUSSION TITLE
Intermittent "...Extension activation failed ..." make GitHub Copilot unavailable

### DISCUSSION THREAD

**Discussion by RunYouCleverBoy** [community/community#125252] [Copilot Conversations] [answered]
### Select Topic Area

Bug

### Body

# The error
Copilot is not functional. "Extension activation failed". No suggestions, no chat. Trying to reset the hosting IDE makes it retry to connect and then time out.

## Environment info
When: Intermittently, several times a day. Began yesterday. Nothing else changed.
Where: MacOS, Visual Studio Code + Android Studio.

## Attempts and observations
* Resetting the IDE (to reset the plugin) makes it show "not signed in".
* Signing in either fails immediately (VS Code) or in Android Studio: stuck forever on "Retrieving GitHub device code". 
<img width="476" alt="image" src="https://github.com/community/community/assets/14019958/31592640-f1a9-42cb-b67e-72336ee93560">

Since it's modal, it blocks my Android Studio. 
Eventually "Failed to initiate the GitHub login process. Please try again."

### Copilot log
On Copilot log (from visual studio code)
```
[error] [auth] Extension activation failed: "connect ETIMEDOUT 3.127.156.149:443"
```

### Other attempts
* settings.json - never touched that.
* Internet connection - checked and working perfectly. No proxy. No special firewalls. Trusted network. Even tried to open a hotspot instead - same.
* Reboot, restart whatever, clear cache - already tried. No luck.
* Solutions online - of course. Tried that, too. Didn't spend more than an hour. You don't expect your an electrician to Google how to fix her/his voltmeter while fiddling with high voltage.
* Github status page - All green.

## Temporary workarounds-ish
Log out, try to login again. Sometimes waits forever and fails. Sometimes after hours - succeeds. Till next incedent.

It takes more time than whatever I gain by using Copilot.

### More attachments
Attached is my Copilot.yaml

It's actually untouched. The bare default.
[Copilot.yml.txt](https://github.com/community/community/files/15410036/Copilot.yml.txt)

## Update (or important clue)
After about 2 hours of doing nothing with Copilot, I tried to restart the IDE's . This time it returned to life (Most of the times it won't).
... Till next time it happens.

Thanks.

**Comment by AmosG** [comment]
having the same issue 
now seems like cant be fixed

**Reply by RunYouCleverBoy** [reply, marked as answer]
I think I resolved it.
In my case it was a tricky one: some genius in my ISP, that decided to block 'api.github.com' thinking it was some kind of a fishy host.

If you ping api.github.com, do you see what blocked it? If it's Bzq - use VPN. Those Einsteins in Bezeq ISP (and that includes some mobile carriers) deployed "bcyber" on some of their DNS instances, and it blocks api.github.com no less.
It works intermittently because for some DNS servers they forgot to install that "feature" so they still work.

**Reply by ibraheemalayan** [reply]
Same issue here, try a VPN or connect to a cellular network ( or anything other than bezeq )

**Reply by ShaiAvr** [reply]
I have the same issue. How do I fix this?? I don't have any VPN or cellular networks.

**Reply by RunYouCleverBoy** [reply]
There are free ones. Google them. I hope our ISP will get sober soon

**Reply by DanielKuris** [reply]
> I think I resolved it. In my case it was a tricky one: some genius in my ISP, that decided to block 'api.github.com' thinking it was some kind of a fishy host.
> 
> If you ping api.github.com, do you see what blocked it? If it's Bzq - use VPN. Those Einsteins in Bezeq ISP (and that includes some mobile carriers) deployed "bcyber" on some of their DNS instances, and it blocks api.github.com no less. It works intermittently because for some DNS servers they forgot to install that "feature" so they still work.

Same here. Same ISP. Downloaded VPN upon viewing this comment and now it's fixed. Good job Bezeq.

**Reply by alifares10** [reply]
did anyone contact bezeq ?

**Reply by shirily** [reply]
same

**Reply by mmfapp** [reply]
It's not fixed for me.
I am trying to contact Bezeq.

**Reply by mmfapp** [reply]
> It's not fixed for me. I am trying to contact Bezeq.

Ok what managed to fix it at least temporarily is turning off cyber protection for the specific device.
As mentioned only a partial solution. we need to talk to Bezeq and they will allow the address of the API to Copilot

Sorry if the English sounds contrived, I'm translating from Hebrew.

**Reply by Mosquid** [reply]
It's super annoying but "הגנת סייבר" can be disabled in BCyber (Be) app in the device settings. This solved the problem for me

**Comment by yehudaTiram** [comment]
I talked to bezeq, and they removed the block. Should be good to all now.

**Reply by Mosquid** [reply]
גיבור

**Reply by aharonhyman** [reply]
i am getting it still, they removed it for a bit but now i am getting it again

**Reply by yehudaTiram** [reply]
Not so fast. Apparently Bezeq  removed the block only on my router, that why my ping is OK. Now I asked them to return it and I am blocked again. see their answer:

לאחר בדיקה כאן עם כל גורמי המקצוע, הוברר שמה שחוסם את השירות התקין זה שכבת ההגנה של הנתב של ה-BECYBER.
נמסר כי מאחר ולאחר תקלה שהייתה ל-Microsoft בוצע תיקון לכתובות שהיו פרוצים ולכן הגנת הסייבר חוסמת את הכתובות,
ממה שאמר לי המומחה שלנו - צריך לפנות לחברת מייקרוסופט כדי לבדוק אם בוצע אימות לכתובת ולכן נחסמו.

חשוב לציין כי מערכתית לצורך  הבדיקה הצלחנו אומנם זמנית לנטרל את רכיב הסייבר,  אך לא ניתן להסיר את הגנת הסייבר לגמרי מאחר והינה חלק בלתי נפרד מהנתב, לא ניתנן לכיבוי או להסרה וכאשר היא כרגע מנוטרלת פותחת את השער לסכנת פגיעות של גורמים זדוניים ופריצות ברשת האינטרנט בתשתית בזק.

**Reply by aharonhyman** [reply]
work around for now is connecting to a hotspot or internet not on bezeq then connect to copilot and then switch back to your regular router until it pings the auth server again anyway

**Reply by RunYouCleverBoy** [reply]
Router thing is, of course, nonsense. It occurs also on hotspot over Pelephone, which, obviously has nothing to do with your home router. Let's wait and see what other "facts"/excuses Bezeq are going to come up with...

**Reply by pavel1860** [reply]
they turned on static IP for me. looks like it fixed the issue.

**Reply by RunYouCleverBoy** [reply]
It won't. 1. you can do that yourself with /private/etc/hosts, 2. It didn't help because it has to go through DNS. I got a Connection Terminated error or so.

**Comment by mmattyg** [comment]
This was blocked at SAM, which Bezeq is using for their cyber security services.
I can confirm this is now resolved and copilot is working again, no VPN required.
